### PR TITLE
reference: inherits is more clear than implies

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -3517,13 +3517,15 @@ more of the closure traits:
 
 * `FnMut`
   : The closure can be called multiple times as mutable. A closure called as
-    `FnMut` can mutate values from its environment. `FnMut` implies
-    `FnOnce`.
+    `FnMut` can mutate values from its environment. `FnMut` inherits from
+    `FnOnce` (i.e. anything implementing `FnMut` should
```
Comment (steveklabnik) [inline src/doc/reference.md]: I'm not sure this is a `should`, I think it happens automatically

File: src/doc/reference.md [hunk 2]
```diff
@@ -3517,13 +3517,15 @@ more of the closure traits:
...
+    also implement `FnOnce`).
 
 * `Fn`
   : The closure can be called multiple times through a shared reference.
     A closure called as `Fn` can neither move out from nor mutate values
-    from its environment. `Fn` implies `FnMut` and `FnOnce`.
+    from its environment. `Fn` inherits from `FnMut`, which itself
+    inherits from FnOnce.
```
Comment (steveklabnik) [inline src/doc/reference.md]: forgot backticks with `FnOnce`

File: src/doc/reference.md (hunk 1 above)
Comment (tshepang) [inline src/doc/reference.md]: removed

File: src/doc/reference.md (hunk 2 above)
Comment (tshepang) [inline src/doc/reference.md]: fixed

**Pull request merged**